config/src/parameters/actual.rs

```diff
@@ -100,27 +126,20 @@ pub enum Genesis {
     },
     /// The peer is responsible for submitting the genesis block
     Full {
-        /// Genesis account key pair
-        key_pair: KeyPair,
-        /// Path to the [`RawGenesisBlock`]
+        /// Genesis account public key
+        public_key: PublicKey,
+        /// Path to the genesis file
         file: PathBuf,
+        /// Genesis signature config with the data to reconstruct signed genesis block
+        signature: GenesisSignatureConfig,
```

Comment:
`parameters::actual` is exposed to the rest of Iroha. Those places don't really need to know about config parsing details.

```suggestion
        signature: iroha_genesis::GenesisSignature,
```